File: 0053-maximum-subarray/0053-maximum-subarray.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) 
    {
        int maxSum = INT_MIN;
        int sum = 0;
        //kadane's Algorithm
        for(int i=0; i<nums.size(); i++)
        {
            sum = max(sum + nums[i], nums[i]);
            maxSum = max(maxSum, sum);
        }
        
        return maxSum;
    }
};

File: 0053-maximum-subarray/0053-maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function(nums) {
    
    //kadane's algorithm
    
    let maximumSum = -Infinity;
    let currentSum = 0;
    
    for(let it of nums)
    {
        currentSum = Math.max(currentSum + it, it);
        maximumSum = Math.max(maximumSum, currentSum);
    }
    return maximumSum;
    
};